js/node/src/session_options_helper.cc

```diff
@@ -23,21 +32,44 @@ void ParseExecutionProviders(const Napi::Array epList, Ort::SessionOptions &sess
   for (uint32_t i = 0; i < epList.Length(); i++) {
     Napi::Value epValue = epList[i];
     std::string name;
+    int deviceId = 0;
+    int coreMlFlags = 0;
     if (epValue.IsString()) {
       name = epValue.As<Napi::String>().Utf8Value();
     } else if (!epValue.IsObject() || epValue.IsNull() || !epValue.As<Napi::Object>().Has("name") ||
                !epValue.As<Napi::Object>().Get("name").IsString()) {
       ORT_NAPI_THROW_TYPEERROR(epList.Env(), "Invalid argument: sessionOptions.executionProviders[", i,
                                "] must be either a string or an object with property 'name'.");
     } else {
-      name = epValue.As<Napi::Object>().Get("name").As<Napi::String>().Utf8Value();
+      auto obj = epValue.As<Napi::Object>();
+      name = obj.Get("name").As<Napi::String>().Utf8Value();
+      if (obj.Has("deviceId")) {
+        deviceId = obj.Get("deviceId").As<Napi::Number>();
+      }
+      if (obj.Has("coreMlFlags")) {
+        coreMlFlags = obj.Get("coreMlFlags").As<Napi::Number>();
+      }
     }
 
     // CPU execution provider
     if (name == "cpu") {
       // TODO: handling CPU EP options
+#ifdef USE_CUDA
     } else if (name == "cuda") {
-      // TODO: handling Cuda EP options
+      Ort::ThrowOnError(OrtSessionOptionsAppendExecutionProvider_CUDA(sessionOptions, deviceId));
```

Comment:
They are not the right functions. 
OrtSessionOptionsAppendExecutionProvider_DML is a global function.  It is deprecated.
The newer ones are SessionOption's member functions. For example, AppendExecutionProvider_CUDA_V2.

I know it is very confusing but I cannot delete the old ones.

Reply:
You can leave it to me.

Reply:
I'm going to build it for testing anyway, so will change in the meantime